### PR TITLE
Iter5

### PR DESCRIPTION
**Pull request by Galionme**

### CODE REVIEW

File: cmd/agent/main.go [hunk 1]
```diff
@@ -1,3 +1,46 @@
 package main
 
-func main() {}
+import (
+	"github.com/Galionme/metric-service.git/cmd/agent/options"
+	"github.com/Galionme/metric-service.git/internal/agent"
```
Comment (arttet) [inline cmd/agent/main.go:5]: CHANGE REQUEST: please sort imports

https://stackoverflow.com/questions/38704769/golang-import-grouping-by-package

File: cmd/agent/options/options.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package options
+
+import (
+	"flag"
+	"fmt"
+	"os"
+)
+
+var (
+	OptionsAgent *Options
```
Comment (arttet) [inline cmd/agent/options/options.go:10]: CHANGE REQUEST: please don't use global variables

File: internal/agent/picker/picker.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	"strconv"
+)
+
+const (
+	bigUint = iota
+	mediumUint
+	bigFloat
+)
+
+type Picker struct {
```
Comment (arttet) [inline internal/agent/picker/picker.go:15]: CHANGE REQUEST: please add a doc comment

File: internal/handlers/home.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"github.com/Galionme/metric-service.git/internal/storage"
+	"html/template"
+	"net/http"
+)
+
+func HomeMetrics(res http.ResponseWriter, req *http.Request) {
+
+	tmpl := `
+  <!DOCTYPE html>
```
Comment (arttet) [inline internal/handlers/home.go:12]: CHANGE REQUEST: it should be in a template file.

File: internal/handlers/value.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+	tmp, ok := storage.GlobalMemStorage.Get(nameMetric)
+	if !ok {
+		res.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	switch typeMetric {
+
+	case "counter":
```
Comment (arttet) [inline internal/handlers/value.go:23]: CHANGE REQUEST: this is a magic string

File: internal/middleware/correctness.go [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package middleware
+
+import (
+	"github.com/go-chi/chi/v5"
+	"net/http"
+)
+
+func CorrectnessType(handler http.Handler) http.Handler {
```
Comment (arttet) [inline internal/middleware/correctness.go:8]: CHANGE REQUEST: it should be a method

File: internal/storage/memStorage.go [hunk 7]
```diff
@@ -0,0 +1,30 @@
+package storage
+
+var GlobalMemStorage *MemStorage
+
+func init() {
```
Comment (arttet) [inline internal/storage/memStorage.go:5]: CHANGE REQUEST: please don't use `init`

**Review by arttet — CHANGES_REQUESTED** [review]

**Comment by arttet** [comment]
Please add the following actions in CI:
- https://about.codecov.io/blog/getting-started-with-code-coverage-for-golang/
- https://golangci-lint.run/usage/install/#ci-installation

File: cmd/agent/main.go [hunk 8]
```diff
@@ -1,3 +1,50 @@
 package main
 
-func main() {}
+import (
+	"github.com/Galionme/metric-service/internal/agent"
```
Comment (arttet) [inline cmd/agent/main.go:4]: CHANGE REQUEST: please sort your imports

https://stackoverflow.com/questions/38704769/golang-import-grouping-by-package

File: cmd/agent/main.go [hunk 9]
```diff
@@ -1,3 +1,50 @@
...
+	"github.com/go-resty/resty/v2"
+)
+
+func main() {
+
+	options := NewOptions()
+
+	err := ParseOptions()
+	if err != nil {
+		panic(err)
```
Comment (arttet) [inline cmd/agent/main.go:16]: CHANGE REQUEST: please don't use `panic`.

File: cmd/agent/main.go [hunk 10]
```diff
@@ -1,3 +1,50 @@
...
+	}
+
+	var cfg config.ConfigAgent
+	if err := env.Parse(&cfg); err != nil {
+		return
+	}
+
+	if cfg.Address != "" && *options.Address != "" {
```
Comment (arttet) [inline cmd/agent/main.go:24]: SUGGESTION: it would be better if it's in `flags.go`

File: cmd/agent/main.go [hunk 11]
```diff
@@ -1,3 +1,50 @@
...
+	service.InitDoctor(*options.ReportInterval, *options.PollInterval)
+}
+
+func sendServer(typeMetric, nameMetric, valueMetric, address string) {
+
+	client := resty.New()
+
+	_, err := client.R().
+		SetHeader("Content-Type", "application/json").
+		Post("http://" + address + "/update/" + typeMetric + "/" + nameMetric + "/" + valueMetric)
```
Comment (arttet) [inline cmd/agent/main.go:44]: SUGGESTION: please use `fmt.Sprint`

File: internal/handlers/update.go [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+func UpdateMetric(res http.ResponseWriter, req *http.Request) {
+
+	nameMetric := chi.URLParam(req, "name")
+	typeMetric := chi.URLParam(req, "type")
+
+	switch strings.ToLower(typeMetric) {
+
+	case "counter":
+
+		tmp, _ := storage.GlobalMemStorage.Get(nameMetric)
```
Comment (arttet) [inline internal/handlers/update.go:21]: CHANGE REQUEST: please don't use global variables

**Review by arttet — CHANGES_REQUESTED** [review]

File: internal/handlers/update.go (hunk 12 above)
Comment (Galionme) [reply internal/handlers/update.go:22]: Я же правильно понял что нужно сделать реализовать сервис?

Comment (arttet) [reply internal/handlers/update.go:22]: You should initialize this service in the entry point.

**Review by arttet — CHANGES_REQUESTED** [review]

**Pull request merged**